lesson_03/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+)
+
+// function that returns an average value of array
+func arrays(s [6]int) float64 {
+	var sum float64 = 0
+	for _, number := range s {
+		x := float64(number)
+		sum += x
+	}
+	n := sum / float64(len(s))
+	return n
+}
+
+// function, that returns the longest word from the slice of strings
+func maxString(words []string) string {
+	var maxWord string
+	for _, word := range words {
+		if len(word) > len(maxWord) {
+			maxWord = word
+		}
+	}
+	return maxWord
+}
+
+// function, that returns the longest word from the slice of strings
+func maxStringBySort(words []string) string {
+	sort.Slice(words, func(i, j int) bool {
+		return len(words[i]) < len(words[j])
+	})
+	s := words[len(words)-1]
+	return s
+}
+
+// function, that returns the copy of the original slice in reverse order
+func reverse(numbers []int64) []int64 {
+	for i := 0; i < len(numbers)/2; i++ {
+		j := len(numbers) - i - 1
+		numbers[i], numbers[j] = numbers[j], numbers[i]
+	}
+	return numbers
+}
+
+// function, that prints map values sorted in order of increasing keys
+func printSorted(m map[int]string) {
+	keys := make([]int, len(m))
+	i := 0
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+	sort.Ints(keys)
+	var keys1 []string
+	for _, k := range keys {
+		keys1 = append(keys1, m[k])
+	}
```

Comment:
there is no reason to create another slice with values only for printing purpose, we can iterate on keys slice and print values.